### PR TITLE
fix(replay): Fix console formatter handling of literal "%"

### PR DESCRIPTION
**Pull request by billyvg**
Use `util.format` (from [browser implementation of node's `util`](https://github.com/browserify/node-util/)) instead of sprintf as it is exactly what we need to parse console.log messages. We do not need the full sprintf formatting support and it also handles literal `%` poorly.

fixes JAVASCRIPT-2AKF
fixes JAVASCRIPT-2EQJ
fixes JAVASCRIPT-2EQK

### CODE REVIEW

**Review by scttcper — APPROVED** [review]
fun fact, jed also includes a full sprintf implementation that does not error on extra `%s`

**Comment by billyvg** [comment]
@scttcper Damn, should we use replace our usage of sprintf with jed's vendored one?

**Comment by scttcper** [comment]
@billyvg maybe? its only 3kb

File: static/app/views/replays/detail/console/messageFormatter.spec.tsx [hunk 1]
```diff
@@ -117,7 +159,7 @@ describe('MessageFormatter', () => {
   it('Should print console message correctly when it is an Error object', () => {
     render(<MessageFormatter breadcrumb={breadcrumbs[2]} />);
 
-    expect(screen.getByText('Error: this is my error message')).toBeInTheDocument();
+    expect(screen.getByText('{}')).toBeInTheDocument();
```
Comment (priscilawebdev) [inline static/app/views/replays/detail/console/messageFormatter.spec.tsx:162]: does it still make sense to have this test? weird that the UI will render `{}` - its not very informative

File: static/app/views/replays/detail/console/messageFormatter.spec.tsx [hunk 2]
```diff
@@ -129,12 +171,34 @@ describe('MessageFormatter', () => {
...
+  // with library sprintfjs this prints undefined, but it ideally should be %s
+  // eslint-disable-next-line
+  it('Should print unbound %s placeholder', () => {
+    render(<MessageFormatter breadcrumb={breadcrumbs[7]} />);
+
+    expect(screen.getByText('Unbound placeholder %s')).toBeInTheDocument();
+  });
+
+  // with library sprintfjs this throws and will just print the string
+  // eslint-disable-next-line
```
Comment (priscilawebdev) [inline static/app/views/replays/detail/console/messageFormatter.spec.tsx:198]: why do we need `eslint-disable-next-line` here?

File: static/app/views/replays/detail/console/messageFormatter.spec.tsx [hunk 3]
```diff
@@ -129,12 +171,34 @@ describe('MessageFormatter', () => {
...
+
+  // with library sprintfjs this prints undefined, but it ideally should be %s
+  // eslint-disable-next-line
+  it('Should print unbound %s placeholder', () => {
+    render(<MessageFormatter breadcrumb={breadcrumbs[7]} />);
+
+    expect(screen.getByText('Unbound placeholder %s')).toBeInTheDocument();
+  });
+
+  // with library sprintfjs this throws and will just print the string
```
Comment (priscilawebdev) [inline static/app/views/replays/detail/console/messageFormatter.spec.tsx:197]: shall we mention that this is fixed now?

File: static/app/views/replays/detail/console/messageFormatter.tsx [hunk 4]
```diff
@@ -25,81 +25,11 @@ function MessageFormatter({breadcrumb}: Props) {
     return <AnnotatedText meta={getMeta(breadcrumb, 'message')} value={logMessage} />;
   }
 
-  // Browser's console formatter only works on the first arg
```
Comment (priscilawebdev) [inline static/app/views/replays/detail/console/messageFormatter.tsx:28]: it is a lot of code removal but I guess this it's better now 🙂

**Review by priscilawebdev — APPROVED** [review]
didn't test locally but the code looks good to me

File: static/app/views/replays/detail/console/messageFormatter.spec.tsx (hunk 1 above)
Comment (billyvg) [reply static/app/views/replays/detail/console/messageFormatter.spec.tsx:162]: Good call, I re-added the special case for this

**Review by ryan953 — APPROVED** [review]

**Pull request merged**